pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -82,9 +123,11 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	// Retry failures are treated the same as Abort.
 	switch hook.FailurePolicy {
 	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
-		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
+		e.event(deployment, kapi.EventTypeWarning, "Failed", fmt.Sprintf("The %s hook failed: %v, aborting deployment %s/%s", label, err, deployment.Namespace, deployment.Name))
```

Comment:
@smarterclayton so you suggesting to make the hook events emitted for replication controller or for deployment config? I prefer the first option, because we already report the Pod creation as RC (deployment) event. So I can see the Pod was created and also see the failure from the command executed in that Pod.